packages/core/workflows-sdk/src/utils/_playground.ts

```diff
@@ -53,6 +55,28 @@ workflow2()
     console.log(res.result)
   })
 
+type MyWorkflowHooks = {
+  beforeCreate: (id: string) => string
+}
+
+const workflowWithHooks = createWorkflowWithHooks(
+  (useHook: UseHook<MyWorkflowHooks>) => {
+    return createWorkflow(
+      "sub-workflow",
+      function (input: WorkflowData<{ outsideWorkflowData: string }>) {
+        step1()
+        step3()
+        const res = useHook("beforeCreate", "id")
+        return step2({ filters: { id: input.outsideWorkflowData } })
+      }
+    )
+  }
+)
+
+workflowWithHooks.beforeCreate((id) => {
```

Comment:
hint: here the handler is also fully typed while maintaining the original workflow types for the arguments